### DISCUSSION TITLE
OnFitView not triggering

### DISCUSSION THREAD

**Discussion by kartlife** [xyflow/xyflow#3185] [Q&A]
Hi guys! First of all, nice library. Now, my problem :( , I am using fitView through reactflow instance to fit view to some nodes. That's ok, but I want to know when fit has endeed, then I am trying to use onFitView to trigger some function, but never trigger. Am I doing well?

Thx in advance

**Comment by bcakmakoglu** [comment]
There is no `onFitView` event you could even listen to, i don't know where you read that 🤔

What exactly are you trying to do and what isn't working for you after calling fit view?

**Reply by kartlife** [reply]
I read it on this page https://reactflow.dev/docs/api/plugin-components/controls/#onfitview . I would  like to fit some elements that are in the canvas, and when this move has endeed, then trigger a function.

**Reply by bcakmakoglu** [reply]
And it's not working to just call the function after you call fitView?

Also, read the description of the prop you linked 😄 
>Description: callback when the fit-view button gets pressed

Nowhere does the description say that this callback is executed when fitView is finished, it's executed when the fit-view *control button* is pressed.

**Reply by kartlife** [reply]
Sorry, i didn`t read well. The problem to call the function after call fitView is that element is still moving due to move duration. Is there any event triggered at the end of node's move. 
Thx for helping me ;)

**Reply by bcakmakoglu** [reply]
Well, the nodes aren't moved at all during fit view - it's the viewport that is moved/zoomed to fit around the nodes at their current positions, so listening for an event after "nodes have moved" won't work in this case.

The only viable solution that comes to mind is using a simple `setTimeout` for the duration you're using for the fit view transition.
So if you use a 500ms transition, just use something like this
```ts
setTimeout(() => {
	// do something after transition
}, 500)
```

Idk if d3 provides any callback for when a transition has ended, so the `fitView` action really just returns `true` or `false` (depending on if fit view was executable) whenever the transition is basically started - so until that is changed, if possible, you'd have to use this sort of "workaround" I'd say 😄
